packages/client/wallets/aa/src/passkeys/PasskeysSDK.ts

```diff
@@ -0,0 +1,163 @@
+import { logError, logInfo, logWarn } from "@/services/logging";
+import type { PasskeyCipher, PasskeysSDKInitParams } from "@/types";
+import { PasskeySdkError, errorToJSON } from "@/utils";
+
+import { BlockchainIncludingTestnet } from "@crossmint/common-sdk-base";
+
+import { CrossmintPasskeyService } from "./services/CrossmintPasskeyService";
+import { LitService } from "./services/LitService";
+
+export class PasskeysSDK {
+    crossmintService: CrossmintPasskeyService;
+    litService: LitService;
+
+    constructor(crossmintService: CrossmintPasskeyService, litService: LitService) {
+        this.crossmintService = crossmintService;
+        this.litService = litService;
+    }
+
+    static init(params: PasskeysSDKInitParams): PasskeysSDK {
+        return new PasskeysSDK(new CrossmintPasskeyService(params.apiKey), new LitService());
+    }
+
+    async signUp(chain: BlockchainIncludingTestnet, walletAddress: string) {
+        try {
+            if (await this.checkCiphersExist(chain, walletAddress)) {
+                logWarn("[SIGN_UP] - PKP and/or Secrets alreay exist. Skipping sign up.", { chain, walletAddress });
+                return;
+            }
+
+            const { pkpEthAddress, pkpPublicKey } = await this.litService.registerWithWebAuthn(
+                `${chain}:${walletAddress}`
+            );
+            await this.savePKP(chain, walletAddress, pkpEthAddress!, pkpPublicKey!);
+
+            logInfo("[PASSKEYS_SIGN_UP] - FINISH", { chain, walletAddress, pkpEthAddress, pkpPublicKey });
+            return { pkpEthAddress, pkpPublicKey };
+        } catch (error: any) {
+            logError("[PASSKEYS_SIGN_UP] - ERROR_SIGNING_UP", { error: errorToJSON(error), chain, walletAddress });
+            throw new PasskeySdkError(`Error signing up [${error?.name ?? ""}]`);
+        }
+    }
+
+    async encryt(chain: BlockchainIncludingTestnet, walletAddress: string, secretToEncrypt: string) {
+        try {
+            const ciphers = await this.getPasskeyCiphers(chain, walletAddress);
+            if (!this.checkPPKAlreadyMinted(ciphers)) {
+                logWarn("[PASSKEYS_ENCRYPT] - PKP not minted. Please sign up first.", { chain, walletAddress });
+                return;
```

Comment:
should we just do this anyway and simplify the API?
Right now we have 3 steps: sign up, encrypt, decrypt
what do you feel about just encrypt, decrypt where encrypt will sign up if needed

Reply:
What I see is that the encrypt will trigger two biometric prompts. Not against the approach, but could feel weird.